src/lib.rs

```diff
@@ -310,13 +312,59 @@ pub fn select_coin_srd(
     })
 }
 
-/// The Global Coinselection API that performs all the algorithms and proudeces result with least [WasteMetric].
+/// The Global Coinselection API that performs all the algorithms and produces result with least [WasteMetric].
 /// At least one selection solution should be found.
+type SelectionFn = fn(&[OutputGroup], CoinSelectionOpt) -> Result<SelectionOutput, SelectionError>;
 pub fn select_coin(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let best_result = Arc::new(Mutex::new(None::<SelectionOutput>));
+
+    // Vector of selection functions
+    let selection_fns: Vec<SelectionFn> =
+        vec![select_coin_fifo, select_coin_lowestlarger, select_coin_srd];
+
+    let handles: Vec<_> = selection_fns
+        .into_iter()
+        .map(|selection_fn| {
+            let best_result_clone = Arc::clone(&best_result);
+            let inputs_clone = inputs.to_vec();
```

Comment:
Is it necessary to clone `inputs`?
None of the functions are mutating the `inputs` directly.
So no need to clone it here